web/js/containers/sidebar/smart-handoff.js

```diff
@@ -158,19 +158,23 @@ class SmartHandoff extends Component {
 
     if (!selectedCollection) return;
 
+    let newBoundaries = boundaries;
+
     const {
       x,
       y,
       width,
       height,
     } = boundaries;
 
-    const newBoundaries = {
-      x,
-      y,
-      x2: x + width,
-      y2: y + height,
-    };
+    if (Object.prototype.hasOwnProperty.call(boundaries, 'width')) {
```

Comment:
I don't think you want to use the [`call` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call) here.  You could check if the `boundaries` object has a `width` property with `Object.prototype.hasOwnProperty` but, I think the simplest way to check if `boundaries.width` is defined would just be to evaluate for truthiness, which should be easy to do inline with the `newBoundaries` object definition:

```
const newBoundaries = {
  ...boundaries
  x2: width ? x + width : boundaries.x2,
  y2: height ? y + height : boundaries.y2,
}
```

Reply:
I was originally just using `Object.prototype.hasOwnProperty` but linter did not like it unless I added the `.call`. I think you are right though, the best way seems to be to do the check within defining the object. I've updated like you suggested above.